psutil/arch/osx/cpu.c

```diff
@@ -109,7 +114,52 @@ psutil_cpu_stats(PyObject *self, PyObject *args) {
     );
 }
 
+#if (defined(__arm64__) && defined(__APPLE__)) || defined(__aarch64__)
+PyObject *
+psutil_cpu_freq(PyObject *self, PyObject *args) {
+    uint32_t min;
+    uint32_t pMin;
+    uint32_t eMin;
+    uint32_t max;
+
+    CFDictionaryRef matching = IOServiceMatching("AppleARMIODevice");
+    io_iterator_t iter;
+
+    IOServiceGetMatchingServices(kIOMainPortDefault, matching, &iter);
+    io_registry_entry_t entry;
+    while ((entry = IOIteratorNext(iter))) {
+        io_name_t name;
+        IORegistryEntryGetName(entry, name);
+
+        if (strncmp(name, "pmgr", 4) == 0) {
+            break;
+        }
+    }
+
+    CFTypeRef pCoreRef = IORegistryEntryCreateCFProperty(entry, CFSTR("voltage-states5-sram"), kCFAllocatorDefault, 0);
+    CFTypeRef eCoreRef = IORegistryEntryCreateCFProperty(entry, CFSTR("voltage-states1-sram"), kCFAllocatorDefault, 0);
+
+    IOObjectRelease(iter);
+    IOObjectRelease(entry);
 
+    size_t pCoreLength = CFDataGetLength(pCoreRef);
+
+    CFDataGetBytes(pCoreRef, CFRangeMake(0, 4), (UInt8 *) &pMin);
+    CFDataGetBytes(eCoreRef, CFRangeMake(0, 4), (UInt8 *) &eMin);
+    CFDataGetBytes(pCoreRef, CFRangeMake(pCoreLength - 8, 4), (UInt8 *) &max);
+
+    min = pMin < eMin ? pMin : eMin;
+
+    CFRelease(pCoreRef);
+    CFRelease(eCoreRef);
+
+    return Py_BuildValue(
+            "sKK",
+            NULL,
```

Comment:
In the x86 world, "current" just returns "max".  That may be preferable here.

Also I'm not a python coder but I would think you'd want the return types identical to the x86 option, e.g., IKK.